demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1062 @@
+
+/**
+ * @brief Date-time to use for the model id
+ */
+#define sampleazureiotDATE_TIME_FORMAT                        "%Y-%m-%dT%H:%M:%S.000Z"
+
+/**
+ * @brief Telemetry values
+ */
+#define sampleazureiotTELEMETRY_NAME                          "temperature"
+
+/**
+ * @brief Property Values
+ */
+#define sampleazureiotPROPERTY_STATUS_SUCCESS                 200
+#define sampleazureiotPROPERTY_SUCCESS                        "success"
+#define sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT        "targetTemperature"
+#define sampleazureiotPROPERTY_MAX_TEMPERATURE_TEXT           "maxTempSinceLastReboot"
+
+/**
+ * @brief Command values
+ */
+#define sampleazureiotCOMMAND_MAX_MIN_REPORT                  "getMaxMinReport"
+#define sampleazureiotCOMMAND_SINCE                           "since"
+#define sampleazureiotCOMMAND_MAX_TEMP                        "maxTemp"
+#define sampleazureiotCOMMAND_MIN_TEMP                        "minTemp"
+#define sampleazureiotCOMMAND_AV_TEMP                         "avgTemp"
+#define sampleazureiotCOMMAND_START_TIME                      "startTime"
+#define sampleazureiotCOMMAND_END_TIME                        "endTime"
+#define sampleazureiotCOMMAND_EMPTY_PAYLOAD                   "{}"
+#define sampleazureiotCOMMAND_FAKE_END_TIME                   "2023-01-10T10:00:00Z"
+
+/**
+ *@brief The Telemetry message published in this example.
+ */
+#define sampleazureiotMESSAGE                                 "{\"" sampleazureiotTELEMETRY_NAME "\":%0.2f}"
+
+/**
+ * @brief Device values
+ */
+#define sampleazureiotDEFAULT_START_TEMP_COUNT                1
+#define sampleazureiotDEFAULT_START_TEMP_CELSIUS              22.0
+#define sampleazureiotDOUBLE_DECIMAL_PLACE_DIGITS             2
+
+/**
+ * @brief The payload to send to the Device Provisioning Service
+ */
+#define sampleazureiotPROVISIONING_PAYLOAD                    "{\"modelId\":\"" sampleazureiotMODEL_ID "\"}"
+
+/**
+ * @brief Time in ticks to wait between each cycle of the demo implemented
+ * by prvMQTTDemoTask().
+ */
+#define sampleazureiotDELAY_BETWEEN_DEMO_ITERATIONS_TICKS     ( pdMS_TO_TICKS( 5000U ) )
+
+/**
+ * @brief Timeout for MQTT_ProcessLoop in milliseconds.
+ */
+#define sampleazureiotPROCESS_LOOP_TIMEOUT_MS                 ( 500U )
+
+/**
+ * @brief Delay (in ticks) between consecutive cycles of MQTT publish operations in a
+ * demo iteration.
+ *
+ * Note that the process loop also has a timeout, so the total time between
+ * publishes is the sum of the two delays.
+ */
+#define sampleazureiotDELAY_BETWEEN_PUBLISHES_TICKS           ( pdMS_TO_TICKS( 2000U ) )
+
+/**
+ * @brief Transport timeout in milliseconds for transport send and receive.
+ */
+#define sampleazureiotTRANSPORT_SEND_RECV_TIMEOUT_MS          ( 2000U )
+
+/**
+ * @brief Transport timeout in milliseconds for transport send and receive.
+ */
+#define sampleazureiotProvisioning_Registration_TIMEOUT_MS    ( 20U )
```

Comment:
this is too small value ... can you updte this to atleast 3 sec... this also need to be changed in other sample too.

Reply:
did it in the other as well